compiler/plugins/target/AMD-AIE/iree-amd-aie/Target/AIETarget.cpp

```diff
@@ -172,7 +172,11 @@ LogicalResult AIETargetBackend::serializeExecutable(
   // generated by `aie2xclbin`
   SmallVector<std::string> entryPointNames;
   for (auto exportOp : variantOp.getExportOps()) {
-    entryPointNames.emplace_back(exportOp.getSymName());
+    // The xclbin kernel name, appended with instance name suffix (`:MLIRAIEV1`,
+    // 10 chars) is required by the xclbinutil to have a length smaller or equal
+    // to 64 right now. To have some additional wiggle room for suffix changes,
+    // we use the 48 first characters for the kernel name.
+    entryPointNames.emplace_back(exportOp.getSymName().substr(0, 48));
```

Comment:
do we need to check if the original size is greater than 44 first?

Reply:
This should be safe: 
```
Returns a substring [pos, pos + count). If the requested substring extends past the end of the string, i.e. the count is greater than size() - pos (e.g. if count == npos), the returned substring is [pos, [size()](https://en.cppreference.com/w/cpp/string/basic_string/size)).
```
from https://en.cppreference.com/w/cpp/string/basic_string/substr